### PR TITLE
Linter rule fixes to maintain parity with SLDS Validator

### PR DESCRIPTION
**Pull request by sf-nithappu**
### Summary
This PR addresses multiple issues with SLDS stylelint rules to improve validator parity and reduce false positives. The changes focus on enhancing the accuracy and reliability of SLDS class validation, token fallback detection, and selector-based rule processing.

####  Bug Fixes
- **Selector Utilities**: Added null check in selector utilities to avoid target properties support issues in selector-based rules
- **SLDS Fallback Rules**: Corrected `slds-no-fallback` rule and deprecated SLDS2 token rules for better accuracy
- **Class Override Detection**: Modified logic to report only the last SLDS class in a selector as override, reducing false positives
- **Token Filtering**: Enhanced fallback detection to stop reporting:
  - `--lwc` tokens in fallback scenarios
  - SDS tokens in fallback scenarios
- **Function Detection**: Improved `t()` function detection logic to avoid reporting when used in fallback contexts

#### 🔧 Refactoring
- Enhanced stylelint rules with property target checks for better rule specificity
- Improved selector-based rule processing with enhanced null safety

### Files Modified
- `packages/stylelint-plugin-slds/src/rules/enforce-bem-usage/index.ts`
- `packages/stylelint-plugin-slds/src/rules/no-deprecated-slds-classes/index.ts` 
- `packages/stylelint-plugin-slds/src/rules/no-slds-class-overrides/index.ts`
- `packages/stylelint-plugin-slds/src/utils/selector-utils.ts`

### Impact
These changes improve the accuracy of SLDS validation by reducing false positives and ensuring proper handling of edge cases in CSS selector processing and token validation.

### CODE REVIEW

**Review by riteshiitbbs007 — APPROVED** [review]

File: packages/stylelint-plugin-slds/tests/rules/no-slds-var-without-fallback/no-slds-var-without-fallback.spec.ts [hunk 1]
```diff
@@ -252,7 +252,7 @@ describe('no-slds-var-without-fallback', () => {
     expect(warningTexts.some(text => text.includes('#c9c9c9'))).toBeTruthy();
   });
 
-  test('Applies fallback value to nested var() functions when fixing', async () => {
+  xtest('Applies fallback value to nested var() functions when fixing', async () => {
```
[the remark this answers is not in the document]
Comment (kenamverma) [reply packages/stylelint-plugin-slds/tests/rules/no-slds-var-without-fallback/no-slds-var-without-fallback.spec.ts:255]: Any reason for xtest here? All others have test.

**Review by kenamverma — APPROVED** [review]

File: packages/stylelint-plugin-slds/tests/rules/no-slds-var-without-fallback/no-slds-var-without-fallback.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (riteshiitbbs007) [reply packages/stylelint-plugin-slds/tests/rules/no-slds-var-without-fallback/no-slds-var-without-fallback.spec.ts:255]: @kenamverma we are targeting only primary value now in var, not the fallback ones.

[the remark this answers is not in the document]
Comment (sf-nithappu) [reply packages/stylelint-plugin-slds/tests/rules/no-slds-var-without-fallback/no-slds-var-without-fallback.spec.ts:255]: @kenamverma  `xtest` helps in keeping the testcase but adds to ignore.

@riteshiitbbs007  Yes, at this moment we want to audit only the main var not vars in fallback to maintain parity with ui-assessment
